### PR TITLE
fix(gsuite): add gsuite model

### PR DESCRIPTION
**Pull request by linuxbandit**
NB this is merged into #183 

some things are to check obviously as I am not aware of their correctness

### CODE REVIEW

File: models/User.js [hunk 1]
```diff
@@ -130,6 +130,14 @@ const User = sequelize.define('user', {
     last_active: {
         type: Sequelize.DATE,
         allowNull: true
+    },
+    gsuite_id: {
```
Comment (WikiRik) [inline models/User.js:134]: No defaultValue here?

File: lib/constants.js [hunk 2]
```diff
@@ -13,6 +13,7 @@ module.exports = {
                 'privacy_consent',
                 'address',
                 'about_me'
+                //does gsuite_id go here?
```
Comment (WikiRik) [inline lib/constants.js:16]: Don't think so, but depends on your implementation of assigning gsuite_id to users on creation.

File: migrations/20201025145040-add-gsuiteID-to-users.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+module.exports = {
+    up: (queryInterface, Sequelize) =>
+      queryInterface.addColumn(
+        'users',
+        'gsuite_id',
+        {
+            type: Sequelize.STRING,
+            allowNull: true,
+            //unique: true
```
Comment (WikiRik) [inline migrations/20201025145040-add-gsuiteID-to-users.js:9]: Why wouldn't it be unique?

File: migrations/20201027145040-add-gsuiteID-to-circles.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
+module.exports = {
+    up: (queryInterface, Sequelize) =>
+      queryInterface.addColumn(
+        'circles',
+        'gsuite_id',
+        {
+            type: Sequelize.STRING,
+            allowNull: true,
+            //unique: true
```
Comment (WikiRik) [inline migrations/20201027145040-add-gsuiteID-to-circles.js:9]: Why wouldn't it be unique?

File: models/Circle.js [hunk 5]
```diff
@@ -16,6 +16,14 @@ const Circle = sequelize.define('circle', {
         validate: {
             notEmpty: { msg: 'Description should be set.' },
         },
+    },
+    gsuite_id: {
```
Comment (WikiRik) [inline models/Circle.js:20]: Why link this to circles and not bodies as well? Might be nice to link the body accounts to the body. And use this solely for linking Google groups to circles

**Review by WikiRik — COMMENTED** [review]
Left some questions

**Comment by WikiRik** [comment]
@serge1peshcoff @linuxbandit any clue on how to do the custom validation? Basically what I want is that it checks the uniqueness for all three (body, circle, user) before accepting it

**Comment by WikiRik** [comment]
I'm moving the custom validation to #183

**Comment by WikiRik** [comment]
@serge1peshcoff @linuxbandit @LeonVreling I guess we can merge this already right? It would not cause any errors

**Review by LeonVreling — APPROVED** [review]

**Pull request merged**